packages/core/src/controllers/verify.ts

```diff
@@ -0,0 +1,86 @@
+import { generateChildLogger, getLoggerContext, Logger } from "@walletconnect/logger";
+import { IVerify } from "@walletconnect/types";
+import { isBrowser } from "@walletconnect/utils";
+import { delay, FIVE_SECONDS } from "@walletconnect/time";
+
+import { VERIFY_CONTEXT, VERIFY_SERVER } from "../constants";
+
+export class Verify extends IVerify {
+  public name = VERIFY_CONTEXT;
+  private verifyUrl: string;
+  private iframe?: HTMLIFrameElement;
+  private initialized = false;
+  private abortController: AbortController;
+
+  constructor(public projectId: string, public logger: Logger) {
+    super(projectId, logger);
+    this.logger = generateChildLogger(logger, this.name);
+    this.verifyUrl = VERIFY_SERVER;
+    this.abortController = new AbortController();
+  }
+
+  public init: IVerify["init"] = async (params) => {
+    // ignore on non browser environments
+    if (!isBrowser() || !document) return;
+
+    this.verifyUrl = params?.verifyUrl || VERIFY_SERVER;
```

Comment:
Since `this.verifyUrl` is set here with fallback, maybe safe to remove it from constructor?

Reply:
it is, but typescript then complains about
- this.verifyUrl is not definitely assigned in the constructor
- requires everywhere to be checked that is assigned i.e.  `?.` on every usage